src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -714,79 +714,91 @@ FileIsLink(char *filename, struct stat filestat)
  * system library's remove_all() method.
  */
 void
-CitusRemoveDirectory(StringInfo filename)
+CitusRemoveDirectory(const char *filename)
 {
-	struct stat fileStat;
-	int removed = 0;
-
-	int fileStated = stat(filename->data, &fileStat);
-	if (fileStated < 0)
-	{
-		if (errno == ENOENT)
-		{
-			return;  /* if file does not exist, return */
-		}
-		else
-		{
-			ereport(ERROR, (errcode_for_file_access(),
-							errmsg("could not stat file \"%s\": %m", filename->data)));
-		}
-	}
-
-	/*
-	 * If this is a directory, iterate over all its contents and for each
-	 * content, recurse into this function. Also, make sure that we do not
-	 * recurse into symbolic links.
-	 */
-	if (S_ISDIR(fileStat.st_mode) && !FileIsLink(filename->data, fileStat))
+	/* files may be added during execution, loop when that occurs */
+	while (true)
 	{
-		const char *directoryName = filename->data;
+		struct stat fileStat;
+		int removed = 0;
 
-		DIR *directory = AllocateDir(directoryName);
-		if (directory == NULL)
+		int fileStated = stat(filename, &fileStat);
+		if (fileStated < 0)
```

Comment:
fileState?

Reply:
So I didn't name this variable. I think it's trying to say "file stat-ed" ie stat worked